lib/ex_unit/lib/ex_unit/manifest.ex

```diff
@@ -16,7 +16,23 @@ defmodule ExUnit.Manifest do
     []
   end
 
+  @spec files_with_failures(t) :: MapSet.t(Path.t())
+  def files_with_failures(manifest) do
+    for({_, entry(last_run_status: :failed, file: file)} <- manifest, do: file, uniq: true)
+    |> MapSet.new()
+  end
+
+  @spec failed_test_ids(t) :: MapSet.t(test_id)
+  def failed_test_ids(manifest) do
+    for({test_id, entry(last_run_status: :failed)} <- manifest, do: test_id, uniq: true)
+    |> MapSet.new()
+  end
```

Comment:
It's sub-optimal that both of these filter to the failed statuses.  If we switch to storing only failures in the manifest (as I explain in my larger comment on the PR), no filtering is needed, and this isn't a concern.  OTOH, if we decide not to implement that optimization, we may want to merge these into one function that returns a tuple so it can filter only once.